consensus/scheme/rolldpos/fsm.go

```diff
@@ -368,14 +368,19 @@ func (m *cFSM) handleStartRoundEvt(_ fsm.Event) (fsm.State, error) {
 	m.ctx.round.proposer = proposer
 	m.ctx.round.timestamp = m.ctx.clock.Now()
 
+	m.produce(m.newCEvt(eInitBlockPropose), 0)
 	// Setup timeout for waiting for proposed block
-	m.produce(m.newCEvt(eInitBlock), 0)
-	m.produce(m.newTimeoutEvt(eProposeBlockTimeout), m.ctx.cfg.AcceptProposeTTL)
+	ttl := m.ctx.cfg.AcceptProposeTTL
+	m.produce(m.newTimeoutEvt(eProposeBlockTimeout), ttl)
+	ttl += m.ctx.cfg.AcceptProposalEndorseTTL
```

Comment:
my understanding is that 2 timeout events are created here in advance, then later can directly return the state such as "sAcceptProposalEndorse" or "sAcceptLockEndorse"

is it possible that consensus does not even move to AcceptProposal or Endorse stage? for instance, the received block is not legit and being rejected

in that case, these 2 events are not necessary right? would that be an issue b/c they are already added to the queue?